### ISSUE TITLE
[3.0] jms child item field serializing

### ISSUE THREAD

**Issue by 7thcubic** [nelmio/NelmioApiDocBundle#1112]
I think the describer is not parsing the child group for jms serializer properly.

Example:

```php
class Foo {
//Groups("content")
//Reference Bar
public $bar;
//Groups("list')
public $id;
}

class Bar {
//Groups("list")
public $id;
//Groups("full")
public $text;
//Groups("say")
public $say;
}
```

with the serialization group of
["list","content", "bar" => "full"]

the serialization registry should handle

```php
$property->setRef(
                    $this->modelRegistry->register(new Model(new Type(Type::BUILTIN_TYPE_OBJECT, false, $type), ["full"]))
                );
```

instead of

```php
$property->setRef(
                    $this->modelRegistry->register(new Model(new Type(Type::BUILTIN_TYPE_OBJECT, false, $type), ["list","content", "bar" => "full"]))
                );
```

Some how or rather it isn't processing correctly right now because we are re-registering the child object with the entire group string which does not generate the correct expected format

https://jmsyst.com/libs/serializer/master/cookbook/exclusion_strategies#overriding-groups-of-deeper-branches-of-the-graph

basically i fix sorta fixed it by doing this though, anyone got a better way of handling it?

```php

                $groups = array_key_exists($name, $model->getGroups()) && is_array($model->getGroups()[$name])? $model->getGroups()[$name] : $model->getGroups();

        $property->setRef(
            $this->modelRegistry->register(new Model(new Type(Type::BUILTIN_TYPE_OBJECT, false, $type), $groups))
         );
```

**Comment by GuilhemN** [comment]
Fixed in https://github.com/nelmio/NelmioApiDocBundle/pull/1132
